lib/fast_jsonapi/serialization_core.rb

```diff
@@ -55,6 +55,14 @@ def ids_hash_from_record_and_relationship(record, relationship)
         id_hash_from_record associated_object, polymorphic
       end
 
+      def fetch_ids(record, relationship)
+        if relationship[:id_from_origin]
```

Comment:
I like this approach. I wonder if this if condition being called every time will have a noticeable performance impact. 

I know the performance tests need a refactor but is it possible to get a sense of the performance impact of this change?

Reply:
@shishirmk Thank you so much for your comment! Let me see how this change affects the performance!